src/librustc_typeck/check/closure.rs

```diff
@@ -32,6 +32,7 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
         debug!("check_expr_closure(expr={:?},expected={:?})",
                expr,
                expected);
+        // FIXME: See if expected_kind here can impact generators
```

Comment:
Nit: please remove. I don't think that's especially relevant to generators, since we only have one generator trait.